src/frontend/sharedComponents/ErrorBottomSheet.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {BottomSheetModal, useBottomSheetModal} from './BottomSheetModal';
+import {Button} from './Button';
+import {StyleSheet, View} from 'react-native';
+import {LogoWithErrorIcon} from './LogoWithErrorIcon';
+import {Text} from './Text';
+import {defineMessages, useIntl} from 'react-intl';
+import {useEffect} from 'react';
+import {useNavigationFromRoot} from '../hooks/useNavigationWithTypes';
+import * as Sentry from '@sentry/react-native';
+
+const m = defineMessages({
+  somethingWrong: {
+    id: 'sharedComponents.ErrorModal.somethingWrong',
+    defaultMessage: 'Something\n Went Wrong',
+  },
+  goBack: {
+    id: 'sharedComponents.ErrorModal.goBack',
+    defaultMessage: 'Go Back',
+  },
+  tryAgain: {
+    id: 'sharedComponents.ErrorModal.tryAgain',
+    defaultMessage: 'Try Again',
+  },
+});
+
+type ErrorModalProps = {
+  error: Error | null;
+  clearError?: () => void;
+};
+
+export const ErrorBottomSheet = ({error, clearError}: ErrorModalProps) => {
+  const {formatMessage} = useIntl();
+  const navigation = useNavigationFromRoot();
+  const {openSheet, sheetRef, isOpen, closeSheet} = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  useEffect(() => {
+    if (error && !isOpen) {
+      Sentry.captureMessage(error.message, 'error');
+      openSheet();
+    }
+  }, [error, isOpen, openSheet]);
```

Comment:
I don't think we need a useEffect here. I think we can do an inline function that can get caught during the render.

```ts
    if (error && !isOpen) {
      Sentry.captureMessage(error.message, 'error');
      openSheet();
    }

```